### PR TITLE
[fix](cluster key) cluster key support vertical_segment_writer

### PR DESCRIPTION
**Pull request by mymeiyi**
The mow cluster key does not support vertical_segment_writer.

### CODE REVIEW

**Comment by mymeiyi** [comment]
run buildall

File: be/src/olap/rowset/segment_v2/vertical_segment_writer.cpp [hunk 1]
```diff
@@ -98,17 +98,40 @@ VerticalSegmentWriter::VerticalSegmentWriter(io::FileWriter* file_writer, uint32
...
+        // encode the rowid into the primary key index
+        if (!_tablet_schema->cluster_key_idxes().empty()) {
+            const auto* type_info = get_scalar_type_info<FieldType::OLAP_FIELD_TYPE_UNSIGNED_INT>();
+            _rowid_coder = get_key_coder(type_info->type());
+            // primary keys
+            _primary_key_coders.swap(_key_coders);
+            // cluster keys
+            _key_coders.clear();
+            _key_index_size.clear();
+            _num_key_columns = _tablet_schema->cluster_key_idxes().size();
```
Comment (dataroaring) [inline be/src/olap/rowset/segment_v2/vertical_segment_writer.cpp:128]: num_key_columns was set as columns of primary key and is set to columns of cluster key. We should add some comment to explain it, comment should be put in .h too.

**Review by zhannngchen — DISMISSED** [review]
LGTM

**Comment by mymeiyi** [comment]
run buildall

**Comment by mymeiyi** [comment]
run buildall

**Comment by mymeiyi** [comment]
run buildall

**Comment by mymeiyi** [comment]
run p0

**Review by dataroaring — APPROVED** [review]
LGTM

**Review by dataroaring — DISMISSED** [review]
LGTM

**Comment by mymeiyi** [comment]
run buildall

**Review by dataroaring — APPROVED** [review]
LGTM

**Review by zhannngchen — APPROVED** [review]
LGTM

**Pull request merged**